pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -359,12 +359,46 @@ func ProcessGetLabelValuesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 			log.Errorf("ProcessGetLabelValuesRequest: Error parsing end time parameter, err:%v", err)
 			return
 		}
+	} else {
+		endTime = uint32(time.Now().Unix())
 	}
 
-	log.Printf("startTime: %v, local time: %v", startTime, time.Unix(int64(startTime), 0).Local())
-	log.Printf("endTime: %v, local time: %v", endTime, time.Unix(int64(endTime), 0).Local())
-	log.Printf("labelName: %v", labelName)
-	// TODO: Implement the logic to get the label values
+	searchText := fmt.Sprintf(`(default_metric{%s="*"})`, labelName)
+	metricQueryRequest, _, _, err := convertPqlToMetricsQuery(searchText, startTime, endTime, myid)
+	if err != nil {
+		ctx.SetContentType(ContentJson)
+		ctx.SetStatusCode(fasthttp.StatusBadRequest)
+		WriteJsonResponse(ctx, nil)
+		log.Errorf("qid=%v, ProcessGetLabelsRequest: Error parsing query err=%+v", qid, err)
+		_, err = ctx.WriteString(err.Error())
+		if err != nil {
+			log.Errorf("qid=%v, ProcessGetLabelsRequest: could not write error message err=%v", qid, err)
+		}
+		return
+	}
+	if len(metricQueryRequest) == 0 {
+		ctx.SetContentType(ContentJson)
+		WriteJsonResponse(ctx, map[string]interface{}{})
+		return
+	}
```

Comment:
we don't need this here right ?

Reply:
actually this len check is fine, ignore my comment, if the len is 0, then we could crash below so good to have this check